### DISCUSSION TITLE
Floating windows Z-order

### DISCUSSION THREAD

**Discussion by matejdro** [i3/i3#5876] [Q&A] [answered]
Whenever I try to copy a file with Dolphin, where destination already exists, I'm greeted with this nice situation:

![image](https://github.com/i3/i3/assets/507922/804e0794-bf00-4d12-9d7f-11296052af9a)

It seems Dolphin first creates destination popup, followed by the copying dialog (that is useless until I confirm the destination popup), which causes windows to be displayed in the wrong z-order:

![animation](https://github.com/i3/i3/assets/507922/b5ed8a52-ac01-4671-b34b-3f76d801fa96)

Is there a way to resolve this situation? Maybe somehow tell i3 to always put windows that match "File already exists" on the top?

Thanks.

**Comment by okraits** [comment, marked as answer]
You could use something like that:

`for_window [title="Copying - Dolphin"] [title="File Already Exists - Dolphin"] focus`

**Reply by matejdro** [reply]
Does i3 actually support multiple criteria? I can't find anything on this in the user's guide. It does not appear to work :/

**Reply by okraits** [reply]
It's not really multiple criteria. The first one is for `for_window` and the second one is for the `focus` command.

I tested it with alacritty and thunar. If a thunar window appears, the focus immediately goes back to the alacritty window.

Please verify that the title strings are correct. I only read them from your screen capture.

**Reply by okraits** [reply]
Another possibility would be to use `no_focus` for the `Copying - Dolphin` window, I guess.

**Reply by matejdro** [reply]
It looks like the rule was conflicting with the following rule that gave focus to every new popup:

`for_window [window_type="dialog"] floating enable focus`

removing `focus`from that rule makes it work. Thanks!
